src/RandomForestBase.js

```diff
@@ -0,0 +1,157 @@
+     * @param {Matrix} trainingSet
+     * @param {Array} trainingValues
+     */
+    train(trainingSet, trainingValues) {
+        if (!Matrix.isMatrix(trainingSet)) {
+            trainingSet = new Matrix(trainingSet);
+        }
+
+        if (Utils.isString(this.options.maxFeatures)) {
+            this.n = Math.floor(functions[this.options.maxFeatures](trainingSet.columns));
+        } else if (Utils.isInt(this.options.maxFeatures)) {
+            if (this.options.maxFeatures > trainingSet.columns) {
+                throw new RangeError('The maxFeatures parameter should be lesser than ' + trainingSet.columns);
+            } else {
+                this.n = this.options.maxFeatures;
+            }
+        } else if (Utils.checkFloat(this.options.maxFeatures)) {
+            this.n = Math.floor(trainingSet.columns * this.options.maxFeatures);
+        } else {
+            throw new RangeError('Cannot process the maxFeatures parameter ' + this.options.maxFeatures);
+        }
+
+
+        if (this.options.classifier) {
+            var Estimator = DTClassfier;
+        } else {
+            Estimator = DTRegression;
+        }
+
+        this.estimators = new Array(this.options.nEstimators);
+        this.indexes = new Array(this.options.nEstimators);
+
+        for (var i = 0; i < this.options.nEstimators; ++i) {
+            var res = Utils.examplesBaggingWithReplacement(trainingSet, trainingValues, this.options.seed);
+            var X = res.X;
+            var y = res.y;
+
+            res = Utils.featureBagging(X, this.n, this.options.replacement, this.options.seed);
+            X = res.X;
+
+            this.indexes[i] = res.usedIndex;
+            this.estimators[i] = new Estimator(this.options.treeOptions);
+            this.estimators[i].train(X, y);
+        }
+    }
+
+    /**
+     * selection method over the predictions of all estimators.
+     * @abstract
+     * @return {Number} prediction.
+     */
+    selection() {
+        throw new Error('Abstract method \'selection\' not implemented!');
+    }
+
+    /**
+     * Predicts the output given the matrix to predict.
+     * @param {Matrix} toPredict
+     * @returns {Array} predictions
+     */
+    predict(toPredict) {
+        var predictionValues = new Array(this.options.nEstimators);
+        for (var i = 0; i < this.options.nEstimators; ++i) {
+            var X = Utils.retrieveFeatures(new Matrix(toPredict), this.indexes[i]);
+            predictionValues[i] = this.estimators[i].predict(X);
+        }
+
+        predictionValues = new Matrix(predictionValues).transpose();
+        var predictions = new Array(predictionValues.length);
+        for (i = 0; i < predictionValues.length; ++i) {
+            predictions[i] = this.selection(predictionValues[i]);
+        }
+
+        return predictions;
+    }
+
+    /**
+     * Export the current model to JSON.
+     * @returns {Object} - Current model.
+     */
+    export() {
```

Comment:
The method should be called `toJSON` (see https://github.com/mljs/ml/blob/master/CONTRIBUTING.md#predictors).
The reason is that calling `JSON.stringify` on the object will automatically (and recursively) use it.